### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by nurbolsultanov**
Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png) ✔️
Set up associations between models ✔️
Created custom methods: ✔️
### ``Users``
- A method that returns the 3 most recent posts for a given user.
### ``Posts``
- A method that updates the posts counter for a user.
- A method that returns the 5 most recent comments for a given post.
### ``Comments``
- A method that updates the comments counter for a post.
### ``Likes``
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
```
Comment (mikethreels) [inline app/models/user.rb]: - [x] in order to make this work you should set the foreign key to 'author_id' for the comments and the likes as well

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (mikethreels) [inline app/models/like.rb]: - [x] after adding the foreign key in the user class, you should change this belongs_to part to look similar to the post class.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (mikethreels) [inline app/models/comment.rb]: - [x] after adding the foreign key in the user class, you should change this belongs_to part to look similar to the post class.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Nurbol, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job starting with the associations.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  belongs_to :post
+
```
Comment (Olamarx) [inline app/models/like.rb]: - [x] Just as the previous reviewer noted, he said you should change it to look like the post.rb belong to and from what I saw in post.rb, there is no foreign_key there, but class_name.

The class_name option is used to specify the name of the class that the association points to.

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
```
Comment (Olamarx) [inline app/models/comment.rb]: - [x] Same thing here.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @everinurmind, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Linters passed ✔️
- Foreign keys set ✔️
- Commit messages ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @everinurmind, 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [x] Foreign keys are set up correctly ✔️ 
- [x] Implemented all the requested changes ✔️ 
- [x] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**